pandas/core/internals/blocks.py

```diff
@@ -2939,18 +2996,25 @@ def copy(self, deep=True):
             values = values.copy(deep=True)
         return self.make_block_same_class(values)
 
-    def external_values(self):
-        """ we internally represent the data as a DatetimeIndex, but for
-        external compat with ndarray, export as a ndarray of Timestamps
-        """
-        return self.values.astype('datetime64[ns]').values
-
     def get_values(self, dtype=None):
+        # TODO: We really need to pin down this type
+        # Previously it was Union[ndarray, DatetimeIndex]
+        # but now it's Union[ndarray, DatetimeArray]
+        # I suspect we really want ndarray, so we need to
+        # check with the callers....
         # return object dtype as Timestamps with the zones
+        # We added an asarray to BlockManager.as_array to work around this.
+        values = self.values
         if is_object_dtype(dtype):
-            return lib.map_infer(
-                self.values.ravel(), self._box_func).reshape(self.values.shape)
-        return self.values
+            values = values._box_values(values._data)
+
+        if self.ndim == 2:
```

Comment:
what?

Reply:
This fixes the bug for `DataFrame.values` on a homogenous 1-column dataframe.

Reply:
use _block_shape

Reply:
Seems like that won't work. `_block_shape` skips EAs https://github.com/pandas-dev/pandas/issues/23023

Reply:
can you add why you need to do this